### PR TITLE
add gpt-3.5-turbo and gpt-4-turbo

### PR DESCRIPTION
**Pull request by matheper**

### CODE REVIEW

File: dln/operator.py [hunk 1]
```diff
@@ -159,7 +161,9 @@ async def _aget_chat_completion_response(self, prompt, **kwargs):
                 )
             except openai.InvalidRequestError as e:
                 # Most likely a content filtering error from Azure.
-                logging.warn(str(e))
+                error_message = f"InvalidRequestError: {e}"
+                logging.warn(error_message)
+                print(colored(error_message, "red"))
                 return str(e)
         else:
```
Comment (MarcCote) [inline dln/operator.py:168]: I think the program should crash if it is not due to a ContentFilter error. I'm fine with simply checking if there's any mention of content filtering in the error message. If not, re-raise the exception.

**Review by MarcCote — CHANGES_REQUESTED** [review]

File: dln/operator.py [hunk 2]
```diff
@@ -146,31 +146,33 @@ def encode(self, string: str) -> List[int]:
...
+            kwargs["model"] = self.engine
+        try:
             response = await openai.ChatCompletion.acreate(
-                model=self.engine,
                 messages=[{"role": "user", "content": prompt}],
                 **kwargs,
             )
+        except openai.InvalidRequestError as e:
+            self._log_filtering_error_message(e)
+            raise
```
Comment (MarcCote) [inline dln/operator.py:175]: Does `raise` the same as `raise e`?

Comment (MarcCote) [reply dln/operator.py:175]: If it is, then all good. Ready to be merged.

Comment (matheper) [reply dln/operator.py]: Not exactly the same, I changed for `raise e` since this includes the original source

**Pull request merged**